comp/core/gui/guiimpl/gui.go

```diff
@@ -228,9 +234,45 @@ func renderIndexPage(w http.ResponseWriter, _ *http.Request) {
 	}
 }
 
+func serveAsset(w http.ResponseWriter, req *http.Request, assetPath string) ([]byte, error) {
+	var rdr io.ReadCloser
+	isGzipped := true
+	rdr, err := viewsFS.Open(assetPath + ".gz")
+	if err != nil && os.IsNotExist(err) {
+		isGzipped = false
+		rdr, err = viewsFS.Open(assetPath)
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer rdr.Close()
+
+	if isGzipped {
+		// if client supports gzip, return as-is
+		if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") {
+			w.Header().Set("Content-Encoding", "gzip")
+		} else {
+			// ungzip on the fly
+			gzReader, err := gzip.NewReader(rdr)
+			if err != nil {
+				return nil, err
+			}
+			// ensure underlying io.Reader is closed
+			defer gzReader.Close()
```

Comment:
Since you are calling `defer rdr.Close()` on line 248. Could we be calling close twice on the same object?

Reply:
From the docs:
> It does not close the underlying [io.Reader](https://pkg.go.dev/io#Reader)

Reply:
I realize the comment needs to be updated